grant-project/grant-monorepo
# Unify template & reduce web3 complexity

Closes #122. Closes #87.

### What This Does

* Combines `AntWrap` and `Web3Page` into a single `Template` component
    * `Template` takes in a `requiresWeb3` boolean that has the same behaviors as Web3Page.
* Moves route configs into an array of config objects that also configure the template.
    * This methodology doesn't work great for dynamic page titles that require fetched resources (e.g. proposal title, user display name etc.) so we'll need a strategy for that.
    * Likewise, this can be used for #131 to configure meta tags for open graph, but will be presented with the same problem for dynamic content.
* Move template render outside of individual routes to prevent flashes of un-rendered pages and prevent needless re-renders.
* Moves web3 bootstrapping logic into a saga, and is only run once initially, rather than by a bunch of components independently.

### Steps to Test

* Browse a bunch of pages, confirm they work fine
* Try to get web3 errors to happen (Wrong network, no metamask etc,) confirm they work fine

---

Note that the git history is pretty messed up here due to some poor rebasing, it's only really ~6 commits.
